developer/ruby/GitHubIssueStats.rb

```diff
@@ -0,0 +1,148 @@
+require 'github_api'
+require 'date'
+require 'yaml'
+
+# Get the begin date, by finding the last major/minor (X.Y.0) release
+# that is at least a week old
+def get_begin_date_and_previous_tag()
+  a_week_ago = Time.now - (60*60*24*7)
+
+  @github.repos.releases.list(owner: @repo_owner,
+                              repo: @repo).each_page do |page|
+    page.each do |release|
+      next if release.tag_name !~ /^v\d\.\d\.0$/
+      release_date = Time.parse(release.created_at)
+      next if release_date > a_week_ago
+      puts "Found previous major/minor release: #{release.tag_name}, #{release_date}"
+      return release_date, " (#{release.tag_name})"
+    end
+  end
+  puts "Cannot find previous release, setting time to 2005"
+  return Time.new(2005, 01, 01), ""
+end
```

Comment:
Instead of harcoding the begin_date, infer it: look at a X.Y.0 release (patch level is 0) that is at least a week old (that should be sufficient to not get the current, freshly created release)